### PR TITLE
[layer] Add support for reduce mean layer

### PR DESCRIPTION
**Pull request by kparichay**
This patch adds support for reduce_mean layer with forwarding and
backwarding implementation.
Basic unittests are added.

~TBD: Golden unittests will come in the next patch.~
Golden unittests added.

Reduce #1722 

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM!

**Comment by zhoonit** [comment]
Could you rebase this to make CI pass? I'll remove `PR/OK2GO` as upstream issue was resolved.

**Comment by kparichay** [comment]
> Could you rebase this to make CI pass? I'll remove `PR/OK2GO` as upstream issue was resolved.

done.

File: nntrainer/layers/reduce_mean_layer.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  This is Reduce Mean Layer Class for Neural Network
+ */
+
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+#include <reduce_mean_layer.h>
+
```
Comment (zhoonit) [inline nntrainer/layers/reduce_mean_layer.cpp:17]: I guess <layer_context.h> should be added here?

File: nntrainer/layers/reduce_mean_layer.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ * @brief  This is Reduce Mean Layer Class for Neural Network
+ *
+ */
+
+#ifndef __REDUCE_MEAN_LAYER_H__
+#define __REDUCE_MEAN_LAYER_H__
+#ifdef __cplusplus
+
+#include <common_properties.h>
+#include <layer_context.h>
```
Comment (zhoonit) [inline nntrainer/layers/reduce_mean_layer.h:19]: Ah it's added here.

File: test/unittest/models/unittest_models.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+#include <neuralnet.h>
+#include <nntrainer_test_util.h>
+
+#include <models_golden_test.h>
+
+using namespace nntrainer;
+
+static inline constexpr const int NOT_USED_ = 1;
+
+static IniSection nn_base("model", "type = NeuralNetwork");
```
Comment (zhoonit) [inline test/unittest/models/unittest_models.cpp:27]: could you make batch_size = 3?

File: test/input_gen/genModelTests_v2.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        out = torch.mean(out, dim=-1)
+        loss = self.loss(out)
+        return out, loss
+
+if __name__ == "__main__":
+    record_v2(
+        ReduceMeanLast(),
+        iteration=2,
+        input_dims=[(2,)],
+        label_dims=[(1,)],
```
Comment (zhoonit) [inline test/input_gen/genModelTests_v2.py:31]: For multiple batches, this should become 

input_dims=[(3, 2)]
label_dims=[[(3, 1)] ?

File: nntrainer/layers/reduce_mean_layer.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/layers/reduce_mean_layer.cpp:17]: https://github.com/nnstreamer/nntrainer/pull/1740#discussion_r760095535

File: test/unittest/models/unittest_models.cpp (hunk 3 above)
Comment (kparichay) [reply test/unittest/models/unittest_models.cpp:27]: done.

File: test/input_gen/genModelTests_v2.py (hunk 4 above)
Comment (kparichay) [reply test/input_gen/genModelTests_v2.py:31]: yes.

**Review by zhoonit — APPROVED** [review]
LGTM :)

**Pull request merged**